src/controller/placeEmergency.controller.ts

```diff
@@ -0,0 +1,93 @@
+import { Request, Response } from "express";
+import {
+  filterHospitalPerDistance,
+  getHospitals,
+} from "../services/places.service";
+import { StatusCode } from "../model/types/StatusCode";
+import { HospitalDTOModel } from "../model/types/models/dto/HospitalDTOModel";
+import { converterHospitaDtoToModel } from "../model/types/models/HospitalsModel";
+
+/**
+ * Busca o hospital mais próximo de acordo com a latitude e longitude informada.
+ * @param req
+ * @param res
+ *
+ * @param latitude: a latitude do usuário
+ * @param longitude: a longitude do usuário
+ * @param range: o alcance para buscar hospitais
+ *
+ * @returns HospitalModel
+ * @throws NotFound
+ */
+async function placeSearchHospitalToEmergencyController(
+  req: Request,
+  res: Response
+) {
+  const latitude = Number(req.query.latitude);
+  const longitude = Number(req.query.longitude);
+  const range = Number(req.query.range);
+
+  if (isNaN(latitude) || isNaN(longitude)) {
+    res.status(404).send("Latitude and Longitude must be valid numbers.");
+    return;
+  }
+
+  const hospitalsList = await getHospitals();
+
+  switch (hospitalsList.status) {
+    case StatusCode.Success: {
+      const hospitalsListFilteredPerDistance = filterHospitalPerDistance(
```

Comment:
Não sei se seria um problema ou algo que ainda não foi adicionado, mas por que não aplicar os filtros diretamente na busca dos hospitais? Acho que isso pode acabar resultando em problemas de performance, principalmente se a lista de hospitais disponíveis for muito grande (primeiro você busca tudo e depois ainda aplica um filtro). Dá uma olhada sobre queries com dados de localização do Mongo https://www.mongodb.com/docs/manual/geospatial-queries/.